ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/RestoreCommandTopicMultipleKafkasIntegrationTest.java

```diff
@@ -0,0 +1,292 @@
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests covering integration tests for backup/restore the command topic when the command topic and streams/tables are on different kafkas.
+ */
+@Category({IntegrationTest.class})
+public class RestoreCommandTopicMultipleKafkasIntegrationTest {
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+  private static final IntegrationTestHarness INTERNAL_TEST_HARNESS = IntegrationTestHarness.build();
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS);
+
+  @ClassRule
+  public static final TemporaryFolder TMP_FOLDER = KsqlTestFolder.temporaryFolder();
+
+  private static File BACKUP_LOCATION;
+  private static TestKsqlRestApp REST_APP;
+  private String commandTopic;
+  private Path backupFile;
+  private Path propertiesFile;
+
+  @BeforeClass
+  public static void classSetUp() throws Exception {
+    BACKUP_LOCATION = TMP_FOLDER.newFolder();
+    INTERNAL_TEST_HARNESS.getKafkaCluster().start();
+
+    REST_APP = TestKsqlRestApp
+        .builder(TEST_HARNESS::kafkaBootstrapServers)
+        .withProperty(KSQL_STREAMS_PREFIX + StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
+        .withProperty(KSQL_METASTORE_BACKUP_LOCATION, BACKUP_LOCATION.getPath())
+        .withProperty(StreamsConfig.STATE_DIR_CONFIG, "/tmp/cat/")
+        .withProperty(KsqlRestConfig.COMMAND_CONSUMER_PREFIX + StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, INTERNAL_TEST_HARNESS.kafkaBootstrapServers())
+        .withProperty(KsqlRestConfig.COMMAND_PRODUCER_PREFIX + StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, INTERNAL_TEST_HARNESS.kafkaBootstrapServers())
+        .build();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    REST_APP.start();
+
+    final KsqlConfig ksqlConfig = new KsqlConfig(REST_APP.getKsqlRestConfig().getKsqlConfigProperties());
+    commandTopic = ReservedInternalTopics.commandTopic(ksqlConfig);
+    backupFile = Files.list(BACKUP_LOCATION.toPath()).findFirst().get();
+    propertiesFile = TMP_FOLDER.newFile().toPath();
+    writeServerProperties(propertiesFile);
+  }
+
+  @After
+  public void teardown() {
+    REST_APP.stop();
+    INTERNAL_TEST_HARNESS.deleteTopics(Collections.singletonList(commandTopic));
+  }
+
+  @After
+  public void teardownClass() {
+    TMP_FOLDER.delete();
+  }
+
+  private static void writeServerProperties(final Path propertiesFile) throws IOException {
+    final Map<String, Object> map = REST_APP.getKsqlRestConfig().getKsqlConfigProperties();
+
+    Files.write(
+            propertiesFile,
+        map.keySet().stream()
+            .map((key -> key + "=" + map.get(key)))
+            .collect(Collectors.joining("\n"))
+            .getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.CREATE
+    );
+  }
+
+  @Test
+  public void shouldBackupAndRestoreCommandTopic() throws Exception {
```

Comment:
Can you add checks to make sure that at the very stages of this test that `TEST_HARNESS` doesn't have the command topic on it?

Reply:
lmk if this is what you invisioned

Reply:
looks good